### PR TITLE
Redesign of xtDesktop theme including addition of Shortcuts and Dashboard

### PR DESCRIPTION
**Pull request by andersdd**
Desktop theme redesign including User Shortcuts, and Dashboard page

Some relevant points:
- Arbitrarily marked as version 4.0.0
- Updated copyright to 2015 on all files
- Main Menu list only displays items if the user has privileges (as opposed to old method of displaying disabled)
- Shortcuts DO display all items, but will give an error dialog if no permissions.  I had to do this due to timing issues with loading extension packages
- Desktop theme is driven by _globalStyle variable string in the initMenu script
- Dashboards require "WebappHostname", "WebappPort", and "desktop/dashboard" metric to be entered.

Initial testing indicates Qt5 likely minimum requirement for Dashboards to work

### CODE REVIEW

**Comment by nedlilly** [comment]
ooh, ooh, super-cool!  Can't wait to hear what's necessary for us regular humans to be able to test this!

**Comment by gilmoskowitz** [comment]
@andersdd The Updater (2.2.5 for Mac) had the following complaints loading the new package:
```
Starting Update at Fri Feb 6 13:56:42 2015
Saving Package Header was successful.
Saving Schema for Package was successful.
Loading new Privileges...
[ trimmed successes ... ]
Completed importing new Privileges.
Applying database change files...
The file database/misc/usersonline-type.sql is empty.

Loading new Function definitions...
The file database/functions/fetchwelcomehtml.sql is empty.

The file database/functions/usersonline.sql is empty.

Completed importing new function definitions.
Loading new MetaSQL Statements...

The MetaSQL statement is empty.

The upgrade has been aborted due to an error and your database was rolled back to the state it was in when the upgrade was initiated.
```

**Comment by andersdd** [comment]
@gilmoskowitz  It was a simple path error in the packaging.  Now resolved.  Also added a fall back page if Dashboard Anything is not available.

**Comment by gilmoskowitz** [comment]
More problems installing into a fresh 4_9_x database (`npm run-script test-build`; note that xtdesktop is already installed):
- the package looks for 4.9.0 or greater but the db is 4.9.0Beta
- tweak the `ServerVersion` metric then get an error that the type `xtdesktop.usersonline` already exists.
- After `drop type ... cascade`: `ERROR:  Cannot create new image ImgUpArrow because another image with that name already exists`

Is Updater 2.2.5 broken?

Dropped xtdesktop schema and trying again... Updater runs without error and the new desktop appears in the Qt client.

**Comment by andersdd** [comment]
Can I please emphasise that the changes made to xtDesktop design are currently technical and this package should not be released until someone (Jen) has worked their graphic design magic over it to actually make it look good.

**Comment by gilmoskowitz** [comment]
@andersdd - got it. it looks pretty good anyway.

**Comment by nedlilly** [comment]
I should point out that I actually "designed" the first OpenMFG logo, including a Microsoft clip art set of blue gears.  So you're in outstanding company ;-)  .... not to say we don't want Jen's ministrations on it!

**Comment by andersdd** [comment]
Hah!!  I heard comments last week about xTuple being designed by engineers....

File: resources/client/scripts/initMenu.js [hunk 1]
```diff
@@ -29,6 +29,12 @@ include("dockSalesHistory");
 include("dockSalesOpen");
 include("dockSendMessage");
 include("dockUserOnline");
+include("desktopMenuBar");
+
+// ================================================
+// Desktop Stylesheet can be maintained from here 
+var _globalStyle = 'font: 75 bold 10pt "Verdana";\ncolor: rgb(138, 138, 138);\nselection-color: rgb(36, 146, 222);\nbackground-color: rgb(255, 255, 255);';
```
Comment (gilmoskowitz) [inline resources/client/scripts/initMenu.js]: While testing a tweak to the revised desktop, the `\n`s here were replaced on Save by the desktop app's script editor with real newlines. The result was a syntax error from the JavaScript engine. Proposed fix:
Replace the string with an array of strings, one element per line of CSS, then `join()` the array with a space:
```JavaScript
var _globalStyle = [
  'font:  ...;',
  'color: ...;',
  ...].join(' ');
```

Comment (andersdd) [inline resources/client/scripts/initMenu.js]: I'm not sure if the \n is needed.  I used the gui client to generate the style and that adds the \n to the text so I just copied and pasted that in.  I will try using the style without the \n and see if it still works.  Simpler than .join()

**Comment by andersdd** [comment]
@gilmoskowitz  New commit resolves version check issues and the other install package bugs, resolves javascript stylesheet element problems.

Adds hotkey refresh of shortcuts menu (using David B's new generic signal).

I packaged, tested and included the new xtdesktop gz file here as well.

**Pull request merged**